prost-build/src/code_generator.rs

```diff
@@ -1145,15 +1169,20 @@ fn build_enum_value_mappings<'a>(
 ) -> Vec<EnumVariantMapping<'a>> {
     let mut numbers = HashSet::new();
     let mut generated_names = HashMap::new();
-    let mut mappings = Vec::new();
+    let mut mappings: Vec<EnumVariantMapping<'a>> = Vec::new();
 
     for (idx, value) in enum_values.iter().enumerate() {
-        // Skip duplicate enum values. Protobuf allows this when the
+        // Remember name for duplicate enum values. Protobuf allows this when the
         // 'allow_alias' option is set.
+        let mut alias_of = None;
         if !numbers.insert(value.number()) {
-            continue;
+            for m in &mappings {
+                if m.proto_number == value.number() {
```

Comment:
`genereated_names` can be used to do this lookup in one step, right?

Reply:
I don't see how, that would require a lookup by value instead of name, no?

Reply:
You are right. The HashMap is the other way around. Please disregard my comment.